### DISCUSSION TITLE
VRoidHubで読み込んだVRM1.0のリップシンクについて

### DISCUSSION THREAD

**Discussion by Natsuki75** [pixiv/vroid-sdk-developers#110] [Q&A]
VRoidHubから読み込んだVRMモデルを取得してリップシンクを出来るようにしたいのですが、表情の口を上書きできるはずのOverrideMouthのblend が全く反映されません。
以下のscriptなのですが、原因などもしお分かりになりましたら教えていただけると幸いです。
また、VRoidSDKで読み込んだモデルをリップシンクする方法についても教えていただけたら嬉しいです。
```cs
using UnityEngine;
using UniVRM10;
using VRoidSDK;

public class VRMEmotionManager : MonoBehaviour
{
    private Vrm10Instance vrm10Instance;
    private ExpressionKey expressionKey;

    public void GetExpression()
    {

        vrm10Instance = GameObject.Find("VRM1").GetComponent<Vrm10Instance>();

        vrm10Instance.Vrm.Expression.Happy.OverrideMouth = UniGLTF.Extensions.VRMC_vrm.ExpressionOverrideType.blend;

        ChangeExpression();
    }

    public void ChangeExpression()
    {
        expressionKey = new ExpressionKey();

        expressionKey = ExpressionKey.CreateFromPreset(ExpressionPreset.happy);
        vrm10Instance.Runtime.Expression.SetWeight(expressionKey, 1.0f);
        expressionKey = ExpressionKey.CreateFromPreset(ExpressionPreset.ee);
        vrm10Instance.Runtime.Expression.SetWeight(expressionKey, 1.0f);
    }
}

```

**Comment by nyamadan** [comment]
> VRoidHubから読み込んだVRMモデルを取得してリップシンクを出来るようにしたいのですが、表情の口を上書きできるはずのOverrideMouthのblend が全く反映されません。

OverrideMouthは、OverrideMouthが設定された表情がアクティブなときに、口に該当する表情（すなわち、 "Aa" ・ "Ih" ・ "Ou" ・ "Ee" ・ "Oh" ）の適用を防ぐものです。
そのため、 "Happy" が適用されている場合、 "Happy" に口のモーフが含まれているか否か（言い換えると、 "Happy" を適用したときに口が開くか否か）に関わらず、口の表情は反映されなくなります。
"Happy" の表情を適用しながらリップシンクによって口を動かしたい場合、モデル側の調整をすることをおすすめします。
OverrideMouthを適用する代わりに、 "Happy" と口の表情を同時に適用しても表情が破綻しないよう、 "Happy" の表情を口を閉じた状態のものにすることが望ましいです。
詳細については[VRMの仕様書](https://github.com/vrm-c/vrm-specification/blob/master/specification/VRMC_vrm-1.0/expressions.ja.md#%E3%83%97%E3%83%AD%E3%82%B7%E3%83%BC%E3%82%B8%E3%83%A3%E3%83%AB%E3%81%AE%E3%82%AA%E3%83%BC%E3%83%90%E3%83%BC%E3%83%A9%E3%82%A4%E3%83%89:~:text=%E4%BE%8B%E3%81%88%E3%81%B0%E3%80%81happy%20%E3%81%8C%20overrideBlink%20%3D%20blend%20%E3%81%AB%E8%A8%AD%E5%AE%9A%E3%81%95%E3%82%8C%E3%81%A6%E3%81%84%E3%82%8B%E5%A0%B4%E5%90%88%E3%80%81%20happy%20%E3%81%AE%E5%80%A4%E3%81%8C%200%20%E3%81%8B%E3%82%89%201%20%E3%81%AB%E3%83%95%E3%82%A7%E3%83%BC%E3%83%89%E3%81%99%E3%82%8B%E3%81%AB%E5%BE%93%E3%81%84%E3%80%81%E7%B7%9A%E5%BD%A2%E3%81%AB%20blink%20%E3%82%92%E6%B8%9B%E8%A1%B0%E3%81%95%E3%81%9B%E3%81%BE%E3%81%99%E3%80%82%200~1%20%E3%81%AE%E9%96%93%E3%81%AE%E4%B8%AD%E9%96%93%E5%80%A4%E3%81%AE%E6%8C%99%E5%8B%95%E3%81%8C%20block%20%E3%81%A8%E7%95%B0%E3%81%AA%E3%82%8A%E3%81%BE%E3%81%99%E3%80%82)に記載があるため、参考にお願いします。

> また、VRoidSDKで読み込んだモデルをリップシンクする方法についても教えていただけたら嬉しいです。

VRoid SDKではリップシンクを行う機能及び、サンプルを提供していません。
そのため、リップシンクを実現する具体的な方法を案内することはできないのですが、以下のページがリップシンク実装の参考になるかもしれないため紹介いたします。
https://tips.hecomi.com/entry/2022/01/30/152519
